repo/fsrepo/fsrepo.go

```diff
@@ -387,6 +389,35 @@ func (r *FSRepo) SetAPIAddr(addr ma.Multiaddr) error {
 	return err
 }
 
+// SetGatewayAddr writes the Gateway Addr to the /gateway file.
+func (r *FSRepo) SetGatewayAddr(addr net.Addr) error {
+	// Create a temp file to write the address, so that we don't leave empty file when the
+	// program crashes after creating the file.
+	f, err := os.Create(filepath.Join(r.path, "."+gatewayFile+".tmp"))
+	if err != nil {
+		return err
+	}
+
```

Comment:
One last basic thing I could spot, you forgot to close the file if the `Fprintf` fails.
All of the return statements are hard to track off, that why you can inject code to run at return time with the `defer` keyword.

Just add:
```go
  defer f.Close()
```
here please.

Reply:
I did not forget that one :)
I followed the API logic. The deletion is in `func (r *FSRepo) Close()`, just like the API one does it too. That should be oke, right? I did test it and the file did get deleted just fine.